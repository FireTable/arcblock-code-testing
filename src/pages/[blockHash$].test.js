import BlockHashPage from './[blockHash$]';
import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

// 对[blockHash$].js进行单元测试   --------------------------------------------------
describe('[blockHash$]页面测试', () => {
  beforeEach(() => {});

  afterEach(() => {});

  test('test snapshot', () => {
    const wrapper = shallow(<BlockHashPage />);
    expect(toJson(wrapper)).toMatchSnapshot(); // 快照
  });

  test('show Waiting for you...', () => {
    const wrapper = mount(<BlockHashPage />);
    const p = wrapper.find('.ant-result-title');
    // 空白页校验
    expect(p.text()).toBe('Waiting for you...');
  });
});

// 对[blockHash$].js进行单元测试   --------------------------------------------------
