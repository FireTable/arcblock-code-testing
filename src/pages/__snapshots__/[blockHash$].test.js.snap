// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[blockHash$]页面测试 show Waiting for you... 1`] = `null`;

exports[`[blockHash$]页面测试 test snapshot 1`] = `
<styled.div>
  <Memo(l) />
  <Result
    icon={<ForwardRef(SmileOutlined) />}
    style={
      Object {
        "margin": "30vh 0",
      }
    }
    subTitle=""
    title="Waiting for you..."
  />
</styled.div>
`;
